routesrv/eskipbytes_test.go

```diff
@@ -0,0 +1,231 @@
+		},
+		{
+			name: "formatAndSet() called with the same route(no change)",
+			routes: [][]*eskip.Route{
+				{
+					{
+						Id: "kube__smth",
+						Headers: map[string]string{
+							"X-Smth": "smth",
+						},
+						Filters: []*eskip.Filter{
+							{
+								Name: filters.RedirectToName,
+								Args: []interface{}{float64(308), "https:"},
+							},
+						},
+						Shunt:       true,
+						BackendType: eskip.ShuntBackend,
+					},
+				},
+				{
+					{
+						Id: "kube__smth",
+						Headers: map[string]string{
+							"X-Smth": "smth",
+						},
+						Filters: []*eskip.Filter{
+							{
+								Name: filters.RedirectToName,
+								Args: []interface{}{float64(308), "https:"},
+							},
+						},
+						Shunt:       true,
+						BackendType: eskip.ShuntBackend,
+					},
+				},
+			},
+			routeString: `kube__smth: Header("X-Smth", "smth") -> redirectTo(308, "https:") -> <shunt>;`,
+			initialised: false,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			// given
+			e := &eskipBytes{useEtag: true}
+			initialised := false
+			// when
+			for _, route := range test.routes {
+				_, initialised = e.formatAndSet(route)
+			}
+			// then
+			assert.Equal(t, test.initialised, initialised)
+			assert.Equal(t, []byte(test.routeString), e.data)
+			assert.Equal(t, strconv.FormatUint(uint64(crc32.ChecksumIEEE([]byte(test.routeString))), 10), e.hash)
+		})
+	}
+}
+
+func TestEskipBytes_ServeHTTP(t *testing.T) {
+	tests := []struct {
+		name      string
+		route     []*eskip.Route
+		Etag      string
+		resBody   string
+		resStatus int
+	}{
+		{
+			name:      "ServeHTTP() called without initialising eSkipBytes",
+			route:     []*eskip.Route{},
+			Etag:      "",
+			resBody:   "",
+			resStatus: http.StatusNotFound,
+		},
+		{
+			name: "ServeHTTP() called after initialising eSkipBytes with empty etag",
+			route: []*eskip.Route{
+				{
+					Id: "kube__smth",
+					Headers: map[string]string{
+						"X-Smth": "smth",
+					},
+					Filters: []*eskip.Filter{
+						{
+							Name: filters.RedirectToName,
+							Args: []interface{}{float64(308), "https:"},
+						},
+					},
+					Shunt:       true,
+					BackendType: eskip.ShuntBackend,
+				},
+			},
+			Etag:      "",
+			resBody:   `kube__smth: Header("X-Smth", "smth") -> redirectTo(308, "https:") -> <shunt>;`,
+			resStatus: http.StatusOK,
+		},
+		{
+			name: "ServeHTTP() called with same Etag hash as eSkipBytes data",
+			route: []*eskip.Route{
+				{
+					Id: "kube__smth",
+					Headers: map[string]string{
+						"X-Smth": "smth",
+					},
+					Filters: []*eskip.Filter{
+						{
+							Name: filters.RedirectToName,
+							Args: []interface{}{float64(308), "https:"},
+						},
+					},
+					Shunt:       true,
+					BackendType: eskip.ShuntBackend,
+				},
+			},
+			Etag:      strconv.FormatUint(uint64(crc32.ChecksumIEEE([]byte(`kube__smth: Header("X-Smth", "smth") -> redirectTo(308, "https:") -> <shunt>;`))), 10),
```

Comment:
Use of CRC is an implementation detail, tests should not rely on any specific value but rather test subsequent request/response with the proper Etag value returned earlier